subsys/bluetooth/controller/include/ll.h

```diff
@@ -9,32 +9,32 @@
 #define LL_ADV_CMDS_LEGACY 1 /* Only legacy advertising cmd/evt allowed */
 #define LL_ADV_CMDS_EXT    2 /* Only extended advertising cmd/evt allowed */
 
-int ll_init(struct k_sem *sem_rx);
+int ll_init(struct k_sem *const sem_rx);
 void ll_reset(void);
 
 uint8_t ll_set_host_feature(uint8_t bit_number, uint8_t bit_value);
 uint64_t ll_feat_get(void);
 
-uint8_t *ll_addr_get(uint8_t addr_type, uint8_t *p_bdaddr);
+uint8_t *ll_addr_get(uint8_t addr_type, uint8_t *const p_bdaddr);
```

Comment:
iirc, `const <type> *ptr` means that whatever `ptr` points to cannot but changed, but the `ptr` value inside the function may still be changed, which doesn't really affect the caller. 

The `<type> *const ptr` means that the value that is being pointed to can be changed, but the `ptr` value cannot be overriden inside the function, but as this doesn't affect the caller in any way, why did you add this to this and many more functions? My assumption would be such that we don't accidentally change the pointer value in function, but is that it?

Reply:
`<type> *const ptr` ensures that function does not change the address supplied by the caller, compiler detects/prevents any bug in the function that could modify contents elsewhere.

Reply:
Right `<type> *const` makes sense to use for `[out]` pointers :)